bindings_generator/src/special_methods.rs

```diff
@@ -1,131 +1,146 @@
-    #[inline]
-    fn class_name() -> &'static str {{
-        "{name}"
-    }}
-
-    #[inline]
-    unsafe fn from_sys(obj: *mut sys::godot_object) -> Self {{
-        {addref_if_reference}
-        Self {{ this: obj, }}
-    }}
-
-    #[inline]
-    unsafe fn from_return_position_sys(obj: *mut sys::godot_object) -> Self {{
-        Self {{ this: obj, }}
-    }}
-
-    #[inline]
-    unsafe fn to_sys(&self) -> *mut sys::godot_object {{
-        self.this
-    }}
-}}
-
-impl ToVariant for {name} {{
-    #[inline]
-    fn to_variant(&self) -> Variant {{ Variant::from_object(self) }}
-}}
-impl FromVariant for {name} {{
-    #[inline]
-    fn from_variant(variant: &Variant) -> Result<Self, FromVariantError> {{ variant.try_to_object_with_error::<Self>() }}
-}}"#,
-        name = class.name,
-        addref_if_reference = if class.is_refcounted() {
-            "object::add_ref(obj);"
-        } else {
-            "// Not reference-counted."
-        },
-    )?;
+    let name = &class.name;
+    let class_name = format_ident!("{}", class.name);
+    let addref_if_reference = if class.is_refcounted() {
+        quote! { object::add_ref(obj); }
+    } else {
+        quote! {
+           // Not reference-counted.
+        }
+    };
+
+    let code = quote! {
+        impl crate::private::godot_object::Sealed for #class_name {}
+
+        unsafe impl GodotObject for #class_name {
+            #[inline]
+            fn class_name() -> &'static str {
+                #name
+            }
+
+            #[inline]
+            unsafe fn from_sys(obj: *mut sys::godot_object) -> Self {
+                #addref_if_reference
+                Self { this: obj, }
+            }
+
+            #[inline]
+            unsafe fn from_return_position_sys(obj: *mut sys::godot_object) -> Self {
+                Self { this: obj, }
+            }
+
+            #[inline]
+            unsafe fn to_sys(&self) -> *mut sys::godot_object {
+                self.this
+            }
+        }
+
+        impl ToVariant for #class_name {
+            #[inline]
+            fn to_variant(&self) -> Variant { Variant::from_object(self) }
+        }
+
+        impl FromVariant for #class_name {
+            #[inline]
+            fn from_variant(variant: &Variant) -> Result<Self, FromVariantError> {
+                variant.try_to_object_with_error::<Self>()
+            }
+        }
+    };
+
+    generated_at!(output);
+    write!(output, "{}", code)?;
 
     Ok(())
 }
 
-pub fn generate_instanciable_impl(output: &mut impl Write, class: &GodotClass) -> GeneratorResult {
-    assert!(class.instanciable);
-
-    writeln!(
-        output,
-        r#"
-impl Instanciable for {name} {{
-    #[inline]
-    fn construct() -> Self {{
-        {name}::new()
-    }}
-}}"#,
-        name = class.name,
-    )?;
+pub fn generate_instantiable_impl(output: &mut impl Write, class: &GodotClass) -> GeneratorResult {
+    if !class.instantiable {
+        return Ok(());
+    }
```

Comment:
If you restore the branches back in `generate_class_bindings`, then this should be removed.

Reply:
It seems like you might have missed this one?